phconvert/pqreader.py

```diff
@@ -643,13 +822,29 @@ def _correct_overflow_nsync(timestamps, detectors, overflow_ch, overflow):
 
 def _correct_overflow_nsync_naive(timestamps, detectors, overflow_ch, overflow):
     """Slow implementation of `_correct_overflow_nsync` used for testing.
-    """
+    """ 
     overflow_correction = 0
     for i in range(detectors.size):
         if detectors[i] == overflow_ch:
             overflow_correction += (overflow * timestamps[i])
         timestamps[i] += overflow_correction
-
+def _t3r_correct_overflow(timestamps, valid, overflow_ch, overflow):
+    """Apply overflow correction when each overflow has a special timestamp.
+    """
+    overflow_correction = 0;
+    for i in xrange(valid.size):
+        if valid[i] == 0:
+            overflow_correction += overflow
+        timestamps[i] += overflow_correction
+def _t3r_correct_overflow2(timestamps, valid, overflow_ch, overflow):
+    """Apply overflow correction when each overflow has a special timestamp.
+    """
+    index_overflows = np.where((valid == 0))
+    num_overflows = timestamps[index_overflows]
+    cum_overflows = np.zeros(timestamps.size, dtype='int64')
+    cum_overflows[index_overflows] = num_overflows
+    np.cumsum(cum_overflows, out=cum_overflows)
+    timestamps += (cum_overflows * overflow)
```

Comment:
This function is very similar to `_correct_overflow_nsync`. The argument `detectors` is renamed `valid` and the argument `overflow_ch` is not used anymore. I suggest just using the previous function. Passing `overflow_ch = 0` should achieve the same effect.

Reply:
The function is removed now.